woocommerce/woocommerce-ios
# Add site ID to `SiteVisitStats` and `TopEarnerStats`: storage layer changes

Storage layer changes for #2916 

## Why

Since we don't delete all data after switching to another store, adding a site ID to data models associated with a store would prevent issues from incorrect data from another store ([example](https://github.com/woocommerce/woocommerce-ios/issues/2916#issuecomment-765296344)).

## Changes

📝 in case the diff count looks scary in this PR: ~2500 diffs are on the mapping model, and ~630 diffs are on the new model!

This PR contains schema changes for the task, the rest of the changes (Yosemite/Networking/app layers) will be in a followup PR.

- Added `siteID` attribute to `SiteVisitStats` and `TopEarnerStats` entity
- Used mapping model: `WooCommerceModelV42toV43.xcmappingmodel` to remove `SiteVisitStats` and `TopEarnerStats` entities in model version 42
- Added a test case for migrating from model version 42 to 43

## Testing

Please confidence check on app migration:

- Launch the app from `develop` or any other branch not based off of PR branch
- Log in to the app if needed
- Launch the app from PR branch --> the app should run as before, the dashboard stats should look the same as before

Update release notes:
